Print prettytable summary for failure

shivahari opened the pull request:
Integrated `prettytable` failure summary to the framework as a `pytest` plugin. The `TerminalReporter` pytest plugin is overwritten with a `CustomTerminalReported` object to print the table summary. The `prettytable` is used only when there is a failure.

Changes tracked in the PR:
1. Created a `pytest` plugin to print the failures in a table using [prettytable](https://pypi.org/project/prettytable/)
2. Modified `conftest.py` file to get the failed scenarios from test objects and pass them on to the `testreporter` plugin
3. Added a  `integrations/reporting_tools/prettytable_object.py` module to configure the table used to print the test failure summary

shivahari commented:
Please ignore the failed `Codacy Static Analysis` check, it complains about the `test_obj` fixture being too complex. We will be fixing it in https://github.com/qxf2/qxf2-page-object-model/issues/471

@rohandudam @Sravanti25

shivahari commented on conftest.py:
```diff
@@ -607,7 +625,21 @@ def pytest_sessionfinish(session, exitstatus):
+        super().__init__(config)
+
+    # Overwrite the summary_failures method to print the prettytable summary
+    def summary_failures(self):
```
@sravantit25 I have added a `try-catch` block  now to catch `broad` exceptions.

shivahari commented on integrations/reporting_tools/prettytable_object.py:
```diff
@@ -0,0 +1,38 @@
+        self.table.field_names = ["Tests Failed"]
+        self.table.align = "l" # <- Align the content of the table left
+
+    def print_table(self, testsummary: dict) -> None:
```
> Would be good to add a check to ensure that testsummary is a dict, else raise ValueError.

This may not be needed. I tried to raise a `TypeError` but looked like an overkill.

rohandudam commented on integrations/reporting_tools/prettytable_object.py:
```diff
@@ -0,0 +1,42 @@
+"""
```
@shivahari @sravantit25 I feel good to move this file to core helper. What do you think???

shivahari replied:
> @shivahari @sravantit25 I feel good to move this file to core helper. What do you think???

This is a custom plugin for `pytest` that overwrites a default `TerminalWriter` object that generally prints the `Failure Summary`. I was thinking this can be considered an integration rather than a core helper since.

shivahari commented:
@rohandudam @sravantit25 I have moved the `prettytable_object` from `integrations/reporting_tools` to `core_helpers` and I have created a module - `custom_pytest_plugins` and added it to `core_helpers`, I moved the `CustomTerminalReporter` object from `conftest` module to `custom_pytest_plugins`.

rohandudam commented:
@shivahari Changes looks good. I ran tests for both positive and negative scenarios. I noticed failure summary is not included in test logs. Can we add it as part of test logs also? If yes, you can take call to make changes as part of this PR or you want to open new one. I am approving PR as of now.

shivahari commented:
> @shivahari Changes looks good. I ran tests for both positive and negative scenarios. I noticed failure summary is not included in test logs. Can we add it as part of test logs also? If yes, you can take call to make changes as part of this PR or you want to open new one. I am approving PR as of now.

The failure summary will not be present in the failure log just like how `pytest` logs are not printed in the test logs currently. This is a `pytest` plugin change.

The summary will appear on the log when we integrate our logging object with `pytest`.
